CA_DataUploaderLib/IOconf/BoardSettings.cs

```diff
@@ -35,19 +36,25 @@ public void SetCalibrationAtIndex(string defaultCalibration, char value, int ind
         public class LineParser
         {
             public static LineParser Default { get; } = new LineParser();
-            private static readonly Regex _hasCommaSeparatedNumbers = new(@"^\s*-?(?:[0-9]*[.])?[0-9]+\s*(?:,\s*-?(?:[0-9]*[.])?[0-9]+\s*)*,?\s*$");
+            private static readonly Regex _hasCommaSeparatedNumbers = new(@"^\s*-?(?:[0-9]*[.])?[0-9]+\s*(?:,\s*-?(?:[0-9]*[.])?[0-9]+\s*)*(?<Status>,\s*0x[0-9a-fA-F]+)?,?\s*$");
 
             /// <returns>the list of doubles, or null when the line did not match the expected format</returns>
-            public virtual List<double>? TryParseAsDoubleList(string line)
+            public virtual (List<double>?, uint) TryParseAsDoubleList(string line)
             {
-                if (!_hasCommaSeparatedNumbers.IsMatch(line))
-                    return null;
+                var status = 0U;
+                var lineMatch = _hasCommaSeparatedNumbers.Match(line);
+                if (!lineMatch.Success)
+                    return (null, status);
 
-                return line.Split(",".ToCharArray())
+                var statusString = lineMatch.Groups["Status"].Value.Trim(',', ' ');
+                if (!string.IsNullOrEmpty(statusString) && statusString.StartsWith("0x"))
```

Comment:
we should use the regex to give you exactly the part of the string you want

something like 

`(?:,\s*0x(?<Status>[0-9a-fA-F]+))?`